src/main/clojure/clara/rules/dsl.clj

```diff
@@ -294,7 +294,43 @@
   ([lhs rhs properties]
      (parse-rule* lhs rhs properties &env)))
 
+;;; added to clojure.core in 1.9
+(defn- qualified-keyword?
+  "Return true if x is a keyword with a namespace"
+  [x] (and (keyword? x) (namespace x) true))
+
+(defn- production-name
+  [name]
```

Comment:
In functions like this (especially), it just seems too risky to shadow `clojure.core/name`. I know it is fully-qualified throughout the body, but just error-prone. 
We could just call the arg `prod-name`

Reply:
Agreed, will fix.